esrally/config.py

```diff
@@ -316,67 +315,51 @@ def create_config(self, config_file, advanced_config=False, assume_defaults=Fals
         if advanced_config:
             root_dir = io.normalize_path(self._ask_property("Enter the benchmark data directory", default_value=root_dir))
         else:
             self.o("* Setting up benchmark data directory in %s" % root_dir)
 
-        if benchmark_from_sources:
-            if not java_10_home or jvm.is_early_access_release(java_10_home):
-                raw_java_10_home = self._ask_property("Enter the JDK 10 root directory", check_path_exists=True, mandatory=False)
-                if raw_java_10_home and jvm.major_version(raw_java_10_home) == 10 and not jvm.is_early_access_release(raw_java_10_home):
-                    java_10_home = io.normalize_path(raw_java_10_home) if raw_java_10_home else None
-                else:
-                    benchmark_from_sources = False
-                    self.o("********************************************************************************")
-                    self.o("You don't have a valid JDK 10 installation and cannot benchmark source builds.")
-                    self.o("")
-                    self.o("You can still benchmark binary distributions with e.g.:")
-                    self.o("")
-                    self.o("  %s --distribution-version=6.0.0" % PROGRAM_NAME)
-                    self.o("********************************************************************************")
-                    self.o("")
-
-        if benchmark_from_sources:
-            # We try to autodetect an existing ES source directory
-            guess = self._guess_es_src_dir()
-            if guess:
-                source_dir = guess
-                logger.debug("Autodetected Elasticsearch project directory at [%s]." % source_dir)
+        if not java_10_home or jvm.is_early_access_release(java_10_home):
+            raw_java_10_home = self._ask_property("Enter the JDK 10 root directory", check_path_exists=True, mandatory=False)
+            if raw_java_10_home and jvm.major_version(raw_java_10_home) == 10 and not jvm.is_early_access_release(raw_java_10_home):
+                java_10_home = io.normalize_path(raw_java_10_home) if raw_java_10_home else None
             else:
-                default_src_dir = os.path.join(root_dir, "src", "elasticsearch")
-                logger.debug("Could not autodetect Elasticsearch project directory. Providing [%s] as default." % default_src_dir)
-                source_dir = default_src_dir
-
-            if advanced_config:
-                source_dir = io.normalize_path(self._ask_property("Enter your Elasticsearch project directory:",
-                                                                  default_value=source_dir))
-            if not advanced_config:
-                self.o("* Setting up benchmark source directory in %s" % source_dir)
+                benchmark_from_sources = False
+                self.o("********************************************************************************")
```

Comment:
These changes should also be reflected in the docs (especially `configuration.rst`). There are further references in `command_line_reference.rst`, `developing.rst`, `faq.rst` and `pipelines.rst` (in the latter case I'd just change the wording in the section `from-sources-complete`, the rest seems fine to me).

Reply:
The latest commit https://github.com/elastic/rally/pull/457/commits/c7ba6a3f1be0bc5b69a026278714b5fc9e7e1646 addresses the mentioned documentation files. I think the JDK / JDK-10 terminology is becoming confusing but luckily we have https://github.com/elastic/rally/issues/452 to fix this.